python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1335,8 +1358,23 @@ def _make_outputs(self, wcsf, visitSummaryTables, exposureInfo):
 
             for d, detector in enumerate(visitSummary["id"]):
                 mapName = f"{visit}/{detector}"
-
-                mapElements = wcsf.mapCollection.orderAtoms(f"{mapName}/base")
+                if mapName in wcsf.mapCollection.allMapNames():
+                    mapElements = wcsf.mapCollection.orderAtoms(f"{mapName}/base")
+                else:
+                    # This extension was not fit, but try to recover WCS
+                    genericElements = mapTemplate["EXPOSURE/DEVICE/base"]["Elements"]
+                    mapElements = []
+                    instrument = visitSummary[0].getVisitInfo().getInstrumentLabel()
+                    for component in genericElements:
+                        elements = mapTemplate[component]["Elements"]
+                        for element in elements:
+                            for generic, specific in {
+                                "BAND": instrument,
+                                "EXPOSURE": visit,
+                                "DEVICE": detector,
+                            }.items():
+                                element = element.replace(generic, str(specific))
```

Comment:
This seems unnecessarily complicated. Why not just do these three things one at a time, so it's easier to follow? Making it a loop like this is just extra lines and harder to debug.

```
element = element.replace("BAND", instrument)
element = element.replace("EXPOSURE", visit)
element = element.replace"DEVICE", detector)
```